addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -7468,6 +7506,31 @@ registry.ImageTools = ImageHandlerOption.extend({
             this._requestUserValueWidgets("remove_img_shape_opt")[0].enable();
         }
     },
+    /**
+     * @override
+     */
+    async _select(previewMode, widget) {
+        await this._super(...arguments);
+        // This is a special case where we need to override the "_select"
+        // function in order to trigger mouse events for hover effects on the
+        // images when previewing the options. This is done here because if it
+        // was done in one of the widget methods, the animation would be
+        // canceled when "_refreshPublicWidgets" is executed in the "_super"
+        if (widget.$el[0].closest("#o_hover_effects_options")) {
+            const hasSetImgShapeHoverEffectMethod = widget.getMethodsNames().includes("setImgShapeHoverEffect");
+            // We trigger the animation when preview mode is "false", except for
+            // the "setImgShapeHoverEffect" option, where we trigger it when
+            // preview mode is "true".
+            if (previewMode === hasSetImgShapeHoverEffectMethod) {
+                this.$target[0].dispatchEvent(new Event("mouseover"));
+                this.hoverTimeoutId = setTimeout(() => {
+                    this.$target[0].dispatchEvent(new Event("mouseout"));
+                }, 700);
+            } else if (previewMode === "reset") {
+                clearTimeout(this.hoverTimeoutId);
```

Comment:
Just to be sure, is it alright to also not trigger `mouseout` here then ? Because it is never triggered then when we clear the timeout on reset (it was in my suggestion but since I did not test I am not sure if it was really necessary)  So just asking to be sure it is alright :slightly_smiling_face:

Reply:
> Just to be sure, is it alright to also not trigger `mouseout` here then ? Because it is never triggered then when we clear the timeout on reset (it was in my suggestion but since I did not test I am not sure if it was really necessary) So just asking to be sure it is alright 🙂

Yes, that's correct :+1: upon reaching this line, the original image has been reinserted into the DOM.